src/db/mod.rs

```diff
@@ -114,6 +114,21 @@ impl FirestoreDb {
         .await
     }
 
+    pub async fn for_default_project_id() -> FirestoreResult<Self> {
+        let firebase_db_options = FirestoreDbOptions::for_default_project_id().await;
+
+        if firebase_db_options.is_some() {
```

Comment:
Instead of using `wrap` and `is_some()` and return for success full path  you could use here

```
if let Some(options) = FirestoreDbOptions::for_default_project_id().await {
   Self::with_options(options).await
}
else {
 Err(...)
}
```

Reply:
Would it be ok if I use a match instead?

Reply:
Would it be ok if I use a match instead?

Reply:
Match is also fine, yes. `unwrap` is a bit if you don't have any other option :)

Reply:
Thank you so much. And sorry for making you lose time, I'm quite new to Rust and certain patterns still need to became part of my routine

Reply:
Don't worry, we all learn new stuff all the time. Thanks for resolving this!